File: ios/Classes/OnUpdatedStreamHandler.swift
import Foundation
import Flutter

class OnUpdatedStreamHandler: NSObject, FlutterStreamHandler{
    var sink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
    
    @objc func onUpdated(value: Bool) {
        guard let sink = sink else { return }
        sink(value)
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

File: android/src/main/kotlin/com/iadvize/flutter_iadvize_sdk/OnUpdatedStreamHandler.kt
package com.iadvize.flutter_iadvize_sdk

import android.os.Handler
import io.flutter.plugin.common.EventChannel
import java.util.*

class OnUpdatedStreamHandler: EventChannel.StreamHandler {
    var sink: EventChannel.EventSink? = null

    fun onUpdated(value: Boolean) {
        sink?.success(value)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }
}